lib/service/Service.ts

```diff
@@ -58,6 +59,8 @@ const argChecks = {
   POSITIVE_AMOUNT: ({ amount }: { amount: number }) => { if (amount <= 0) throw errors.INVALID_ARGUMENT('amount must be greater than 0'); },
   POSITIVE_QUANTITY: ({ quantity }: { quantity: number }) => { if (quantity <= 0) throw errors.INVALID_ARGUMENT('quantity must be greater than 0'); },
   PRICE_NON_NEGATIVE: ({ price }: { price: number }) => { if (price < 0) throw errors.INVALID_ARGUMENT('price cannot be negative'); },
+  QUANTITY_CORRECT_PRECISION: ({ quantity }: { quantity: number }) => { if (checkDecimals(quantity)) throw errors.INVALID_ARGUMENT('quantity must have less than 13 decimal'); },
+  PRICE_CORRECT_PRECISION: ({ price }: { price: number }) => { if (checkDecimals(price)) throw errors.INVALID_ARGUMENT('price must have less than 13 decimal'); },
```

Comment:
Let's go with `PRICE_MAX_DECIMAL_PLACES` for the check name and `price cannot have more than 12 decimal places` for the message, if you don't mind.